pkg/inventory/inventory-client.go

```diff
@@ -79,79 +62,157 @@ func NewInventoryClient(factory util.Factory) (*ClusterInventoryClient, error) {
 	if err != nil {
 		return nil, err
 	}
-	builder := factory.NewBuilder()
+	builderFunc := factory.NewBuilder
 	clusterInventoryClient := ClusterInventoryClient{
-		builder:                   builder,
-		mapper:                    mapper,
-		validator:                 validator,
-		pastInventoryObjects:      []*resource.Info{},
-		retrievedInventoryObjects: false,
+		builderFunc: builderFunc,
+		mapper:      mapper,
+		validator:   validator,
+		clientFunc:  factory.UnstructuredClientForMapping,
+		dryRun:      false,
 	}
 	return &clusterInventoryClient, nil
 }
 
-// GetStoredObjRefs returns the set of previously applied objects as ObjMetadata,
-// or an error if one occurred. This set of previously applied object references
-// is stored in the inventory objects living in the cluster. Uses information
-// from the current inventory object to find the previously applied inventory
-// objects.
-func (cic *ClusterInventoryClient) GetStoredObjRefs(currentInv *resource.Info) ([]object.ObjMetadata, error) {
-	prevInventories, err := cic.GetPreviousInventoryObjects(currentInv)
+// Merge stores the union of the passed objects with the objects currently
+// stored in the cluster inventory object. Retrieves and caches the cluster
+// inventory object. Returns the set differrence of the cluster inventory
+// objects and the currently applied objects. This is the set of objects
+// to prune. Creates the initial cluster inventory object storing the passed
+// objects if an inventory object does not exist. Returns an error if one
+// occurred.
+func (cic *ClusterInventoryClient) Merge(localInv *resource.Info, objs []object.ObjMetadata) ([]object.ObjMetadata, error) {
```

Comment:
Merge is also calculating diff and returning pruneIds, maybe a better name which reflects entire behavior of method would be great.

Reply:
This is a large, long PR. Let' keep this as is for now.